docs/programming/using_bobs.md

```diff
@@ -0,0 +1,149 @@
+copProcessBlocks();
+vPortWaitForEnd(s_pVp);
+```
+
+## What are BOBs?
+
+BOBs are 2D graphic elements that can be efficiently moved around the screen using the Amiga's hardware blitter. They offer several advantages:
+
+- Background preservation for clean undrawing
+- Transparency support via masks
+- Animation capabilities by switching frames
+- Efficient memory usage and blitter operations
+- Support for double-buffering for smooth animation
+
+## Basic BOB Workflow
+
+The typical workflow for using BOBs is:
+
+1. **Initialization** (in gamestate create):
+```c
+// Create manager with your buffers
+bobManagerCreate(pFrontBuffer, pBackBuffer, uwAvailHeight);
+
+// Initialize each BOB
+bobInit(&sBob1, uwWidth, uwHeight, 1, pFrameData, pMaskData, uwX, uwY);
+bobInit(&sBob2, uwWidth, uwHeight, 1, pFrameData, pMaskData, uwX, uwY);
+// ... initialize more BOBs as needed
+
+// Allocate memory for background saving (must be called after all BOBs are initialized)
+bobReallocateBuffers();
+```
+
+2. **Game Loop** (in gamestate loop):
+```c
+// Begin the BOB drawing cycle - undraws previous BOBs
+bobBegin();
+
+// Update BOB positions
+sBob1.sPos.uwX = newX;
+sBob1.sPos.uwY = newY;
+
+// Add BOBs to drawing queue
+bobPush(&sBob1);
+// ... perform other calculations
+bobPush(&sBob2);
+
+// Process some BOBs (save backgrounds) while doing other calculations
+bobProcessNext();
+
+// Signal no more BOBs will be added this frame
+bobPushingDone();
+
+// Process remaining BOBs (draw them)
+while(bobProcessNext()) {}
+// Or simply call bobEnd() to finish all processing
+bobEnd();
+```
+
+3. **Cleanup** (in gamestate destroy):
+```c
+bobManagerDestroy();
+```
+
+## Important BOB Functions
+
+### Initialization and Setup
+- `bobManagerCreate()`: Initializes the BOB system with front/back buffers
+- `bobInit()`: Sets up a BOB with its dimensions, position, and frame data
+- `bobReallocateBuffers()`: Allocates memory for background saving
+
+### Animation and Dimensions
+- `bobSetFrame()`: Changes a BOB's animation frame
+- `bobSetWidth()`: Modifies a BOB's width (be careful with memory allocation)
+- `bobSetHeight()`: Modifies a BOB's height (be careful with memory allocation)
+- `bobCalcFrameAddress()`: Helps calculate the memory address of a specific animation frame
+
+### Drawing Cycle
+- `bobBegin()`: Starts a new drawing cycle, undraws previous BOBs
+- `bobPush()`: Adds a BOB to the drawing queue
+- `bobProcessNext()`: Processes the next BOB operation (background save or drawing)
+- `bobPushingDone()`: Signals no more BOBs will be added this frame
+- `bobEnd()`: Completes all drawing operations
+- `bobDiscardUndraw()`: Skips undrawing (useful for first frame)
+
+## Tips and Best Practices
+
+1. **Memory Management**:
+   - For BOBs with background preservation (isUndrawRequired=1), the initial width/height should be set to the maximum anticipated size.
+   - Changing width/height to larger than initial size can cause memory corruption.
```

Comment:
I'd move it somewhere near describing `bobReallocateBuffers()` and remove this section

```suggestion
> [!NOTE]
> When using the undraw buffers mode, for BOBs with background preservation (`isUndrawRequired` set to `1`) and varying width/height, the initial width/height should be set to the maximum anticipated size. Failing to do so will result in memory corruption.
```